src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+/* ****************************************************************************
+*
+* subTreat - 
+*/
+static void subTreat(std::string tenant, BSONObj& bobj)
+{
+  BSONElement  idField = bobj.getField("_id");
+
+  if (idField.eoo() == true)
+  {
+    LM_E(("Database Error (error retrieving _id field in doc: '%s')", bobj.toString().c_str()));
+    return;
+  }
+
+
+
+  //
+  // 01. Extract values from database object 'bobj'
+  //
+
+  std::string               subId         = idField.OID().toString();
+  int64_t                   expiration    = bobj.getField("expiration").Long();
+  std::string               reference     = bobj.getField("reference").String();
+  int64_t                   throttling    = bobj.hasField("throttling") ? bobj.getField("throttling").Long() : -1;
+  std::vector<BSONElement>  eVec          = bobj.getField("entities").Array();
+  std::vector<BSONElement>  attrVec       = bobj.getField("attrs").Array();
+  std::vector<BSONElement>  condVec       = bobj.getField("conditions").Array();
+  std::string               formatString  = bobj.hasField("format") ? bobj.getField("format").String() : "XML";
+  std::string               servicePath   = bobj.hasField("servicePath") ? bobj.getField("servicePath").String() : "/";
+  Format                    format        = stringToFormat(formatString);
+
+  std::vector<EntityInfo*> eiV;
+  std::vector<std::string> attrV;
+  Restriction              restriction;
+  NotifyConditionVector    notifyConditionVector;
+
+  //
+  // 02. Pushing Entity-data names to EntityInfo Vector (eiV)
+  //
+  for (unsigned int ix = 0; ix < eVec.size(); ++ix)
+  {
+    BSONObj entity = eVec[ix].embeddedObject();
+
+    if (!entity.hasField("id"))
+    {
+      LM_W(("Runtime Error (got a subscription without id)"));
+      continue;
+    }
+
+    std::string id = entity.getStringField("id");
+    
+    if (!entity.hasField("isPattern"))
+    {
+      continue;
+    }
+
+    std::string isPattern = entity.getStringField("isPattern");
+    if (isPattern != "true")
+    {
+      continue;
+    }
+
+    std::string  type = "";
+    if (entity.hasField("type"))
+    {
+      type = entity.getStringField("type");
+    }
+
+    EntityInfo* eiP = new EntityInfo(id, type);
+    eiV.push_back(eiP);
+  }
+
+  if (eiV.size() == 0)
+  {
+    return;
+  }
+
+
+  //
+  // 03. Pushing attribute names to Attribute Vector (attrV)
+  //
+  for (unsigned int ix = 0; ix < attrVec.size(); ++ix)
+  {
+    std::string attributeName = attrVec[ix].String();
+
+    attrV.push_back(attributeName);
+  }
+
+
+  //
+  // 04. FIXME P4: Restriction not implemented
+  //
+  
+  
+
+  //
+  // 05. Fill in notifyConditionVector from condVec
+  //
+  for (unsigned int ix = 0; ix < condVec.size(); ++ix)
+  {
+    BSONObj                   condition = condVec[ix].embeddedObject();
+    std::string               condType;
+    std::vector<BSONElement>  valueVec;
+
+    condType = condition.getStringField("type");
+    if (condType != "ONCHANGE")
```

Comment:
This confirms that only ONCHANGE conditions are added (related with a previous comment about a ONCHANGE chekcing in the match logic that I think is not actually needed).

NTC